start-tests.sh

```diff
@@ -7,97 +7,209 @@
+  export OPTIONS_HUB_OC_IDP=kube:admin
+else
+  log_color "purple" "OPTIONS_HUB_OC_IDP" "detected, using $OPTIONS_HUB_OC_IDP for test.\n"
+fi
 
-# Check to see if CYPRESS_OC_IDP is null.
-if [[ -z $CYPRESS_OC_IDP || "$CYPRESS_OC_IDP" == "null" ]]; then
-  log_color "purple" "CYPRESS_OC_IDP" "is (null or not set); setting to 'kube:admin'\n"
-  export CYPRESS_OC_IDP=kube:admin
+# Check to see if OPTIONS_HUB_BASEDOMAIN, OPTIONS_HUB_USER, or OPTIONS_HUB_PASSWORD are missing. We need these to run the UI test.
+if [[ -z $OPTIONS_HUB_BASEDOMAIN || -z $OPTIONS_HUB_USER || -z $OPTIONS_HUB_PASSWORD ]]; then
+  log_color "red" "One or more exported variables are undefined for hub cluster." "(set ${PURPLE}OPTIONS_HUB_BASEDOMAIN, OPTIONS_HUB_BASEDOMAIN, and OPTIONS_HUB_BASEDOMAIN${NC} to execute the test with environment variables)\n"
+
+  if [ ! -f $OPTIONS_HUB_KUBECONFIG ]; then
+    log_color "red" "The kubeconfig file for hub cluster was not located." "(set ${PURPLE}KUBECONFIG${NC} to ${YELLOW}$OPTIONS_HUB_KUBECONFIG${NC} and oc login to create kubeconfig file."
+    exit 1
+  else
+    if [[ -z $OPTIONS_HUB_PASSWORD || "$OPTIONS_HUB_PASSWORD" == "null" ]]; then
+      log_color "purple" "OPTIONS_HUB_PASSWORD" "is required to login into the ACM console. ${YELLOW}Skipping UI test.${NC}\n"
+      export SKIP_UI_TEST=true
+    fi
+
+    echo -e "Kubeconfig file detected at: $OPTIONS_HUB_KUBECONFIG => copying to ./cypress/config/hub-kubeconfig"
+    cp $OPTIONS_HUB_KUBECONFIG ./config/hub-kubeconfig
+    export OPTIONS_HUB_KUBECONFIG=./config/hub-kubeconfig
+
+    if [[ -z $OPTIONS_HUB_KUBECONTEXT || "$OPTIONS_HUB_KUBECONTEXT" == "null" ]]; then
+      HUB_CLUSTER=($(oc config get-clusters --kubeconfig=$OPTIONS_HUB_KUBECONFIG))
+      export OPTIONS_HUB_KUBECONTEXT=default/${HUB_CLUSTER[1]}/kube:admin
+    fi
+
+    echo -e
+    log_color "cyan" "Switching context to log into Kube API server"
+    oc config use-context --kubeconfig=$OPTIONS_HUB_KUBECONFIG $OPTIONS_HUB_KUBECONTEXT
+    
+    export OPTIONS_HUB_BASEDOMAIN=$(oc whoami --show-server=true | cut -d'.' -f2- | cut -d':' -f1)
+
+    if [[ $OPTIONS_HUB_OC_IDP == "kube:admin" && -z $OPTIONS_HUB_USER ]]; then
+      export OPTIONS_HUB_USER=kubeadmin
+    fi
+
+    log_color "purple" "HUB CLUSTER:" "$OPTIONS_HUB_BASEDOMAIN"
+  fi
+else
+  echo -e "Environment variables detected. Configuring tests to execute with exported variables."
 fi
 
+# Export variables to cypress.
+export CYPRESS_OPTIONS_HUB_BASEDOMAIN=$OPTIONS_HUB_BASEDOMAIN
+export CYPRESS_OPTIONS_HUB_KUBECONFIG=$OPTIONS_HUB_KUBECONFIG
+export CYPRESS_OPTIONS_HUB_KUBECONTEXT=$OPTIONS_HUB_KUBECONTEXT
+export CYPRESS_OPTIONS_HUB_PASSWORD=$OPTIONS_HUB_PASSWORD
+export CYPRESS_OPTIONS_HUB_USER=$OPTIONS_HUB_USER
+export CYPRESS_OPTIONS_HUB_OC_IDP=$OPTIONS_HUB_OC_IDP
+
+# Export base url for cluster.
+export BASE_URL=https://multicloud-console.apps.$OPTIONS_HUB_BASEDOMAIN
+export CYPRESS_BASE_URL=$BASE_URL
+
+echo -e
```

Comment:
Export all of the `CYPRESS_OPTIONS_HUB_*` variables here. This will just make sure we're capturing the right value regardless of the testing process.